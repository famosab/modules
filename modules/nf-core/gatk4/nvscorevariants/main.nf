process GATK4_NVSCOREVARIANTS {
    tag "$meta.id"
    label 'process_low'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://community-cr-prod.seqera.io/docker/registry/v2/blobs/sha256/b2/b28daf5d9bb2f0d129dcad1b7410e0dd8a9b087aaf3ec7ced929b1f57624ad98/data':
        'community.wave.seqera.io/library/gatk4_gcnvkernel:e48d414933d188cd' }"

    input:
    tuple val(meta), path(vcf), path(tbi), path(aligned_input), path(intervals)
    path fasta
    path fai
    path dict
    path architecture
    path weights

    output:
    tuple val(meta), path("*nv.vcf.gz")    , emit: vcf
    tuple val(meta), path("*nv.vcf.gz.tbi"), emit: tbi
    path "versions.yml"                    , emit: versions


    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    def aligned_input_cmd = aligned_input ? "--input $aligned_input" : ""
    def interval_command = intervals ? "--intervals $intervals" : ""
    def architecture_cmd = architecture ? "--architecture $architecture" : ""
    def weights_cmd = weights ? "--weights $weights" : ""

    def avail_mem = 3072
    if (!task.memory) {
        log.info '[GATK NVScoreVariants] Available memory not known - defaulting to 3GB. Specify process memory requirements to change this.'
    } else {
        avail_mem = (task.memory.mega*0.8).intValue()
    }
    """
    gatk --java-options "-Xmx${avail_mem}M -XX:-UsePerfData" \\
        NVScoreVariants \\
        --variant $vcf \\
        --output ${prefix}.nv.vcf.gz \\
        --reference $fasta \\
        $interval_command \\
        $aligned_input_cmd \\
        $architecture_cmd \\
        $weights_cmd \\
        --tmp-dir . \\
        $args


    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        gatk4: \$(samtools --version |& sed '1!d ; s/samtools //')
    END_VERSIONS
    """

    stub:
    def prefix = task.ext.prefix ?: "${meta.id}"
    """
    echo | gzip -c > ${prefix}.cnn.vcf.gz
    touch ${prefix}.cnn.vcf.gz.tbi

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        gatk4: \$(echo \$(gatk --version 2>&1) | sed 's/^.*(GATK) v//; s/ .*\$//')
    END_VERSIONS
    """
}
