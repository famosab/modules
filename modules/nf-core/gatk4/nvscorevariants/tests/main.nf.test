nextflow_process {

    name "Test Process GATK4_NVSCOREVARIANTS"
    script "../main.nf"
    process "GATK4_NVSCOREVARIANTS"

    tag "modules"
    tag "modules_nfcore"
    tag "gatk4"
    tag "gatk4/nvscorevariants"

    test("homo sapiens - vcf") {
        when {
            process {
                """
                input_vcf = [
                    [ id:'test' ], // meta map
                    file(params.modules_testdata_base_path + "genomics/homo_sapiens/illumina/gvcf/test.genome.vcf.gz", checkIfExists: true),
                    file(params.modules_testdata_base_path + "genomics/homo_sapiens/illumina/gvcf/test.genome.vcf.gz.tbi", checkIfExists: true),
                    [],
                    []
                ]
                fasta  = file(params.modules_testdata_base_path + "genomics/homo_sapiens/genome/genome.fasta", checkIfExists: true)
                fai    = file(params.modules_testdata_base_path + "genomics/homo_sapiens/genome/genome.fasta.fai", checkIfExists: true)
                dict   = file(params.modules_testdata_base_path + "genomics/homo_sapiens/genome/genome.dict", checkIfExists: true)

                input = [input_vcf, fasta, fai, dict, [], []]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(
                    process.out.versions,
                    path(process.out.vcf[0][1]).vcf.variantsMD5,
                    file(process.out.tbi[0][1]).name,
                ).match() }
            )
        }

    }

    test("homo sapiens - vcf - stub") {

        options "-stub"

        when {
            process {
                """
                input_vcf = [
                    [ id:'test' ], // meta map
                    file(params.modules_testdata_base_path + "genomics/homo_sapiens/illumina/gvcf/test.genome.vcf.gz", checkIfExists: true),
                    file(params.modules_testdata_base_path + "genomics/homo_sapiens/illumina/gvcf/test.genome.vcf.gz.tbi", checkIfExists: true),
                    [],
                    []
                ]
                fasta  = file(params.modules_testdata_base_path + "genomics/homo_sapiens/genome/genome.fasta", checkIfExists: true)
                fai    = file(params.modules_testdata_base_path + "genomics/homo_sapiens/genome/genome.fasta.fai", checkIfExists: true)
                dict   = file(params.modules_testdata_base_path + "genomics/homo_sapiens/genome/genome.dict", checkIfExists: true)

                input = [input_vcf, fasta, fai, dict, [], []]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(
                    process.out,
                    path(process.out.versions[0]).yaml
                    ).match() }
            )
        }

    }

}
