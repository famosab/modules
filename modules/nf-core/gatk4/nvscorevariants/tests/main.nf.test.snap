{
    "homo sapiens - vcf - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test.nv.vcf.gz:md5,68b329da9893e34099c7d8ad5cb9c940"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test"
                        },
                        "test.nv.vcf.gz.tbi:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "2": [
                    "versions.yml:md5,53ff464ab460a1bc3b5916d21e69a080"
                ],
                "tbi": [
                    [
                        {
                            "id": "test"
                        },
                        "test.nv.vcf.gz.tbi:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "vcf": [
                    [
                        {
                            "id": "test"
                        },
                        "test.nv.vcf.gz:md5,68b329da9893e34099c7d8ad5cb9c940"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,53ff464ab460a1bc3b5916d21e69a080"
                ]
            },
            {
                "GATK4_NVSCOREVARIANTS": {
                    "gatk4": "NVSCOREVARIANTS-PREVIEW-SNAPSHOT"
                }
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.6"
        },
        "timestamp": "2025-05-28T17:05:30.233766162"
    }
}